test/e2e/multihoming.go

```diff
@@ -507,6 +614,14 @@ func namespacedName(namespace, name string) string {
 }
 
 func inRange(cidr string, ip string) error {
+	// cidr is in the form of either <subnet>/<prefix> or <subnet>/<prefix>/<nodePrefix>
+	s := strings.Split(cidr, "/")
+	if len(s) > 3 || len(s) < 2 {
+		return fmt.Errorf("invalid cidr %s", cidr)
+	} else if len(s) == 3 {
+		// this is cidr of layer3 network
+		cidr = s[0] + "/" + s[1] 
+	}
```

Comment:
I prefer to do the opposite: have the caller always pass a valid CIDR to this function. 

I'm doing something like this in https://github.com/ovn-org/ovn-kubernetes/pull/3442/files#diff-8a0101c1714fa2f18fcb47cc2f71b4a820bc6e88fb23f0fcb557abafbfb3b924R201 

Let's see what the maintainers think.

Reply:
Yup, certainly better to have a different specific function to parse the CIDR from `<subnet>/<prefix>/<nodePrefix>` to be used independently from this `InRange` function.

Reply:
fixed